net/adoptopenjdk/bumblebench/string/StringConversionBench.java

```diff
@@ -34,362 +35,90 @@
-	private static final boolean IS_DEBUG = option("debug", false);
-	private static final boolean IS_VERBOSE = option("verbose", false);
-	private static final boolean IS_VERIFY = option("verifyResult", true);
-	private static final boolean IS_BYTE_ARRAY_TEST = option("byteArrayTest",
-			false);
+	private static String[] strings;
+	private static char[] possibleChars = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l',
+		'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E',
+		'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
+		'Y', 'Z', ' ', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
 
 	static {
-		System.out.println("Testing "
-				+ ((IS_TESTING_TO_UPPER) ? "toUpper" : "toLower"));
+		strings = new String[MAX_NUM_STRINGS];
+		rand = new Random();
+		int length;
+		StringBuilder sb = new StringBuilder(1000);
 
-		if (MAX_CHAR > 0x0ff) {
-			System.out
-					.println("this bench is not able to handle max_char > 0xff, clippig it to 0xff");
+		if (!IS_PSUEDO_RANDOM) {
+			rand.setSeed(12345L);
 		}
-	}
-
-	// dump converted char to a file
-	private static final boolean IS_DUMP_CONVERTED = option(
-			"dumpConvertedChar", false);
-
-	private int stringLength = 51;
-	private int numString = 256; // number of strings to convert per iteration
-	private String _escape; // To prevent the optimizer from removing the whole
-							// method
-
-	private byte[] _escapeByte;
-
-	protected int maxIterationsPerLoop() {
-		return MAX_NUM_STRINGS;
-	}
-
-	private void printArrays(byte[] before, byte[] after) {
-		System.out.println("before: " + Arrays.toString(before));
-		System.out.println("after: " + Arrays.toString(after));
-	}
-
-	// this will be a recognized function handled by the codegen directly
-	// for compressed string conversion test
-	public byte[] byteArrayToUpperWrapper(byte[] input, byte[] output) {
-		return byteArrayToUpper(input, output);
-	}
-
-	public byte[] byteArrayToLowerWrapper(byte[] input, byte[] output) {
-		return byteArrayToLower(input, output);
 
-	}
-
-	private final byte[] byteArrayToUpper(byte[] input, byte[] output) {
-		return null;
-	}
-
-	private final byte[] byteArrayToLower(byte[] input, byte[] output) {
-		return null;
-	}
-
-	/**
-	 * This temporary hack function is intended to test the JIT's ability to do
-	 * case conversion on byte arrays. each byte (0x00 to 0xff) is a char in the
-	 * ascii world.
-	 * */
-	private long runByteArrayTest() {
-		long counter = 0;
-		Random rand = new Random();
-		int numString = 256;
-		int stringLength = 0;
-
-		File diffFile = new File("./conversionDiff");
-		PrintWriter writer = null;
-
-		if (IS_DUMP_CONVERTED) {
-			if (!diffFile.exists()) {
-
-				try {
-					writer = new PrintWriter(diffFile);
-				} catch (FileNotFoundException e) {
-					e.printStackTrace();
+		if (STRING_LENGTH <= 0  || STRING_LENGTH > 1000) {
+			for (int i = 0; i < MAX_NUM_STRINGS; i++) {
```

Comment:
enhanced for loops or collections / streams not possible here?

Reply:
I don't think enhanced for loops could be used here since i'm assigning new values to each element of `strings`. I think using `Stream.map()` here would be possible but what would the benefit be?

Reply:
None :-), was just checking assumptions :-)